src/Host/Client/Impl/Definitions/IRSessionProvider.cs

```diff
@@ -5,19 +5,28 @@
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.R.Host.Client.Host;
 
 namespace Microsoft.R.Host.Client {
     public interface IRSessionProvider : IDisposable {
-        IRSession GetOrCreate(Guid guid);
+        IRSession GetOrCreate(Guid guid, IRHostBrokerConnector brokerConnector);
```

Comment:
We probably may want to import IRSessionProvider to the RInteractiveWorkflow, and then define properties for REPL session, Intellisense session, packages, etc.

---
In reply to: [73038600](https://github.com/Microsoft/RTVS/pull/2133#discussion_r73038600) [](ancestors = 73038600,73035733)